### PR TITLE
update nrosy to match the MIQ paper

### PR DESCRIPTION
**Pull request by hankstag**
[Describe your changes and what you've already done to test it.]
The current nrosy implementation is not consistent with the MIQ paper, and the singularity index has wrong sign. This PR is to fix this.

#### Check all that apply (change to `[x]`)
- [ ] All changes meet [libigl style-guidelines](https://libigl.github.io/style-guidelines/).
- [ ] Adds new .cpp file.
- [ ] Adds corresponding unit test.
- [ ] Adds corresponding python binding.
- [x] This is a minor change.

### CODE REVIEW

**Comment by danielepanozzo** [comment]
This looks ok, but it does not pass the build.

File: include/igl/copyleft/comiso/nrosy.cpp [hunk 1]
```diff
@@ -494,32 +494,32 @@ void igl::copyleft::comiso::NRosyField::computek()
...
 
-      assert(V1(fid1_vc,2) < 10e-10);
-      assert(V1((fid1_vc+1)%3,2) < 10e-10);
+      assert(V1(fid1_vc,2) < 1e-10);
+      assert(V1((fid1_vc+1)%3,2) < 1e-10);
 
       // compute rotation R such that R * N1 = N0
       // i.e. map both triangles to the same plane
-      double alpha = -std::atan2(V1((fid1_vc + 2) % 3, 2), V1((fid1_vc + 2) % 3, 1));
+      double alpha = -std::atan2(-V1((fid1_vc + 2) % 3, 2), -V1((fid1_vc + 2) % 3, 1));
```
Comment (jiangzhongshi) [inline include/igl/copyleft/comiso/nrosy.cpp]: What difference does the sign make here?

File: include/igl/copyleft/comiso/nrosy.cpp [hunk 2]
```diff
@@ -528,17 +528,25 @@ void igl::copyleft::comiso::NRosyField::computek()
...
+        return (0 == y) ? x : x - y * floor(x/y);
+      };
+      ktemp = pos_fmod(ktemp, 2*igl::PI);
+      if (ktemp > igl::PI) 
+        ktemp -= 2*igl::PI;
+      
       // just to be sure, rotate ref0 using angle ktemp...
       Eigen::MatrixXd R2(2,2);
-      R2 << std::cos(ktemp), -std::sin(ktemp), std::sin(ktemp), std::cos(ktemp);
+      R2 << std::cos(-ktemp), -std::sin(-ktemp), std::sin(-ktemp), std::cos(-ktemp);
```
Comment (jiangzhongshi) [inline include/igl/copyleft/comiso/nrosy.cpp]: This line looks strange, should it be equivalent to 
`R2 << std::cos(ktemp), std::sin(ktemp), -std::sin(ktemp), std::cos(ktemp);`

File: include/igl/copyleft/comiso/nrosy.cpp (hunk 1 above)
Comment (jdumas) [reply include/igl/copyleft/comiso/nrosy.cpp]: If you flip both points coordinates you should get a rotation or pi.

**Pull request merged**